### PR TITLE
Import projects base imp

### PR DESCRIPTION
**Pull request by labhvam5**

### CODE REVIEW

File: apps/mappings/exceptions.py [hunk 1]
```diff
@@ -50,3 +51,60 @@ def new_fn(workspace_id: int, *args):
...
+        }
+        try:
+            return func(expense_attribute_instance, *args)
+        except WrongParamsError as exception:
+            error['message'] = exception.message
+            error['response'] = exception.response
+            error['alert'] = True
+            import_log.status = 'FAILED'
+
+        except (QBOCredential.DoesNotExist):
```
Comment (ashwin1111) [inline apps/mappings/exceptions.py:75]: can merge QBOCredential.DoesNotExist, QBOWrongParamsError, QBOInvalidTokenError together

File: apps/mappings/queues.py [hunk 2]
```diff
@@ -107,26 +109,25 @@ def construct_tasks_and_chain_import_fields_to_fyle(workspace_id):
     mapping_settings = MappingSetting.objects.filter(workspace_id=workspace_id, import_to_fyle=True)
     credentials = QBOCredential.objects.get(workspace_id=workspace_id)
 
-    task_settings = {
+    task_settings: TasksSettings = {
```
Comment (ashwin1111) [inline apps/mappings/queues.py:112]: `TaskSetting`, let's make dataclasses should singular, the variable can be plural - that's okay

File: apps/mappings/queues.py [hunk 3]
```diff
@@ -107,26 +109,25 @@ def construct_tasks_and_chain_import_fields_to_fyle(workspace_id):
...
-                'destination_field': mapping_setting.destination_field,
-                'destination_sync_method': SYNC_METHODS[mapping_setting.destination_field],
-                'is_auto_sync_enabled': get_auto_sync_permission(mapping_setting),
-                'is_custom': False,
-            })
-
-    # Make a call to the SDK
-    # chain_import_fields_to_fyle(workspace_id, task_settings)
-    pass
+    if mapping_settings:
```
Comment (ashwin1111) [inline apps/mappings/queues.py:122]: if is not required

File: apps/mappings/queues.py [hunk 4]
```diff
@@ -107,26 +109,25 @@ def construct_tasks_and_chain_import_fields_to_fyle(workspace_id):
...
+        for mapping_setting in mapping_settings:
+            if mapping_setting.source_field in ['PROJECT']:
```
Comment (ashwin1111) [inline apps/mappings/queues.py:124]: I remember same check is there in imports too, if 1 place solves it, let's keep it in API.

Comment (labhvam5) [reply apps/mappings/queues.py:124]: this is api only. 🤣

Comment (ashwin1111) [reply apps/mappings/queues.py:124]: no no, remove the other one if keeping in just API should work

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**